File: ruby_small_problems/easy_4/6.rb
# takes an array of numbers
# returns an array with the same number of elements, 
# so each element has the running total from the orig array

def running_total(arr)
  total = 0
  arr.map { |x| total += x }
end

puts running_total([2, 5, 13]) == [2, 7, 20]
puts running_total([14, 11, 7, 15, 20]) == [14, 25, 32, 47, 67]
puts running_total([3]) == [3]
puts running_total([]) == []

File: 210/lecture/practice_problems/arrays/intermediate_array_uses/6.js
function uniqueElements(arr) {
  let i;
  let newArr = [];
  for (i = 0; i < arr.length; i += 1) {
    if (!newArr.includes(arr[i])) {
      newArr.push(arr[i]) 
    }
  }
  return newArr;
}

console.log(uniqueElements([1, 2, 4, 3, 4, 1, 5, 4]));
// returns [1, 2, 4, 3, 5]
